packages/build-tools/src/builders/android.ts

```diff
@@ -77,55 +67,18 @@ async function buildAsync(ctx: BuildContext<Android.Job>): Promise<void> {
   });
 
   await ctx.runBuildPhase(BuildPhase.RESTORE_CACHE, async () => {
-    await ctx.cacheManager?.restoreCache(ctx);
-    if (ctx.shouldRestoreCache) {
-      try {
-        const cacheKey = await generateCacheKeyAsync(workingDirectory);
-        const jobId = nullthrows(ctx.env.EAS_BUILD_ID, 'EAS_BUILD_ID is not set');
-
-        const robotAccessToken = nullthrows(
-          ctx.job.secrets?.robotAccessToken,
-          'Robot access token is required for cache operations'
-        );
-        const expoApiServerURL = nullthrows(
-          ctx.env.__API_SERVER_URL,
-          '__API_SERVER_URL is not set'
-        );
-
-        const { archivePath } = await downloadCacheAsync({
-          logger: ctx.logger,
-          jobId,
-          expoApiServerURL,
-          robotAccessToken,
-          paths: cachePaths,
-          key: cacheKey,
-          keyPrefixes: [CACHE_KEY_PREFIX],
-          platform: ctx.job.platform,
-        });
-
-        await decompressCacheAsync({
-          archivePath,
-          workingDirectory,
-          verbose: ctx.env.EXPO_DEBUG === '1',
-          logger: ctx.logger,
-        });
-
-        ctx.logger.info('Cache restored successfully');
-        await asyncResult(
-          spawnAsync('ccache', ['--zero-stats'], {
-            env: ctx.env,
-            logger: ctx.logger,
-            stdio: 'pipe',
-          })
-        );
-      } catch (err: unknown) {
-        if (err instanceof TurtleFetchError && err.response.status === 404) {
-          ctx.logger.info('No cache found for this key. Create a cache with function save_cache');
-        } else {
-          ctx.logger.warn({ err }, 'Failed to restore cache');
-        }
-      }
+    if (ctx.isLocal) {
+      ctx.logger.info('Restore cache is not supported for local builds');
```

Comment:
Suggestion to adjust it as you've adjusted `ios.ts` or move to `restoreCcacheAsync` as you've suggested.